tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -22,6 +24,132 @@ readonly KBS_NS="coco-tenant"
 # The kbs service name
 readonly KBS_SVC_NAME="kbs"
 
+# Set resource data.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
+#	$4 - resource data
+#
+# Note: kbs-client will be installed if not yet.
+#
+kbs_set_resource() {
+	local repository="${1:-}"
+	local type="${2:-}"
+	local tag="${3:-}"
+	local data="${4:-}"
+	local file
+	local rc=0
+
+	if [ -z "$data" ]; then
+		echo "ERROR: missing data parameter"
+		return 1
+	fi
+
+	file=$(mktemp -t kbs-resource-XXXXX)
+	echo "$data" > "$file"
+
+	kbs_set_resource_from_file "$repository" "$type" "$tag" "$file" || \
+		rc=$?
+
+	rm -f "$file"
+	return $rc
+}
+
+# Set resource, read data from file.
+#
+# Parameters:
+#	$1 - repository name (optional)
+#	$2 - resource type (mandatory)
+#	$3 - tag (mandatory)
+#	$4 - resource data
+#
+# Note: kbs-client will be installed if not yet.
+#
+kbs_set_resource_from_file() {
+	local repository="${1:-}"
+	local type="${2:-}"
+	local tag="${3:-}"
+	local file="${4:-}"
+
+	if [[ -z "$type" || -z "$tag" ]]; then
+		echo "ERROR: missing type='$type' and/or tag='$tag' parameters"
+		return 1
+	elif [ ! -f "$file" ]; then
+		echo "ERROR: resource file '$file' does not exist"
+		return 1
+	fi
+
+	local path=""
+	[ -n "$repository" ] && path+="${repository}/"
+	path+="${type}/"
+	path+="${tag}"
+
+	kbs_install_cli
+
+	kbs-client --url "$(kbs_k8s_svc_http_addr)" config \
+		--auth-private-key "$(kbs_get_private_key)" set-resource \
+		--path "$path" --resource-file "$file"
+}
+
+# Return the private key file used for CLI authentication.
+#
+kbs_get_private_key() {
+	echo "$COCO_KBS_DIR/kbs/config/kubernetes/base/kbs.key"
```

Comment:
This is a fixed string, wouldn't it be better to set it as a variable, rather than to execute an echo every time we ask for private key path?